rust/src/core/polars/mod.rs

```diff
@@ -184,4 +184,8 @@ impl DPNamespace {
             .dp()
             .laplace(scale)
     }
+
+    pub fn mean<L: Literal>(self, bounds: (L, L), scale: f64) -> Expr {
+        self.0.clone().dp().sum(bounds, scale) / self.0.len()
```

Comment:
In this case it's copying even less- just an expression, which has no data inside it. Later on, Polars does common sub-expression elimination to combine these two branches back together (as well as any other computation that's duplicated). 

LazyFrames are also cheap to clone. If they have data inside them (were created by calling df.lazy()) then the data is behind a reference-counted pointer. Otherwise lazyframes just hold metadata about the computation.